src/interfaces/docker_interface.py

```diff
@@ -157,8 +157,9 @@ def get_messages(self, a_name: str, b_ipv4: str, bitcoin_network: str = "regtest
         return messages
 
 
-    def logs_grep(self, pattern: str, container_name: str):
+    def logs_grep(self, pattern: str, network: str):
         compiled_pattern = re.compile(pattern)
+        container_name = self.get_container(f"{network}_fluentd").name
```

Comment:
assumes we are always using fluentd for the logger. also, would be really nice if anywhere we see a hardcoded string (`_fluentd`, in this case). in my WIP k8s branch I've had to hunt down all these places and modify these strings because k8s doesn't give the same names as docker compose.

Reply:
This is in the docker interface though? WYD in here on k8s?

Reply:
on the k8s interface, the fluentd service has a slightly different name so i just updated it inline. but my comment was more a general: "hardcoded strings are codesmell, we should be abstracting this into a config, and not specifying them in the interfaces"

that way, if something changes (e.g. we change the logging on the docker interface, or the naming changes on the k8s interface, etc), we have a single config file where were update the name once and then the rest of the code "just works"

Reply:
Hmmm, I guess I don't understand why your in this file still? But I agree re. hardcoded strings.

My master plan is actually to rip out fluentd, see middle commits here https://github.com/bitcoin-dev-project/warnet/pull/173/commits

It seems to use a lot of resources and so doesn't scale. Docker's stdout logging can work for us in this sense, so it might be best plan to just use whatever is native/easiest in k8s too? As it's an interface method you can do whatever you like with it on your k8s side.

I just changed how I grep all logs in the followup commit: [`55ff125` (#173)](https://github.com/bitcoin-dev-project/warnet/pull/173/commits/55ff125810c875a20a6186cd753d5581adebd62e)

I think the fact the log parser is in Ruby is the culprit, I often saw that maxing out one of my precious cores...

Reply:
I was trying to get fluentd working on k8s, and IIRC ended up coping some stuff over from the docker interface. But agree that fluentd isn't great (i never did end up getting it working on k8s), and we will likely end up using something else for k8s. The comment was less about fluentd and more about pointing out places where we use string names in specific functions. Not critical to fix at this point, but more a "wouldn't it be nice" as we go forward.